src/goto-instrument/contracts/utils.cpp

```diff
@@ -448,9 +448,19 @@ void generate_history_variables_initialization(
   program.destructive_append(history);
 }
 
+bool is_transformed_loop_head(const goto_programt::const_targett &target)
+{
+  // The head of a transformed loop is
```

Comment:
is this just

return is_assignment_to_instrumented_variable(target, ENTERED_LOOP) &&
           (target->assign_rhs() == false_exprt());

?

Seems like a more straightforward way to say it.